File: internal/mcp/testdata/effect_contexts_4821/sync_service.py
def sync(self, request):
    contacts = Contact.objects.filter(active=True)
    if request.data.get("force"):
        Contact.objects.create(name="seed")
    for contact in contacts:
        requests.post("https://api.example.com/notify", json={"id": contact.id})
    return Response({"ok": True})

File: internal/mcp/testdata/effect_contexts_4821/sync.service.ts
export class SyncService {
  async sync(req: Request) {
    const rows = await this.repo.find();
    if (req.body.force) {
      await this.repo.save({ name: 'seed' });
    }
    for (const row of rows) {
      await fetch('https://api.example.com/notify', { method: 'POST' });
    }
    return { ok: true };
  }
}
